### ISSUE TITLE
How I can apply batch normalization?

### ISSUE THREAD

**Issue by fmaglia** [dennybritz/cnn-text-classification-tf#29]
I read this paper: https://arxiv.org/abs/1502.03167 and I would to test in my model the technique.

I found the implementation for tensorflow 0.9 (function batch_norm(...)):
https://github.com/tensorflow/tensorflow/blob/r0.9/tensorflow/contrib/layers/python/layers/layers.py

but how I can apply this at the code?

**Comment by dhwajraj** [comment]
There are many ways to achieve this, pointing one of those here. 
in text_cnn.py replace following lines:
```
W = tf.Variable(tf.truncated_normal(filter_shape, stddev=0.1), name="W")
b = tf.Variable(tf.constant(0.1, shape=[num_filters]), name="b")
conv = tf.nn.conv2d(
                    self.embedded_chars_expanded,
                    W,
                    strides=[1, 1, 1, 1],
                    padding="VALID",
                    name="conv")
# Apply nonlinearity
h = tf.nn.relu(tf.nn.bias_add(conv, b), name="relu")
```
to this:

```
conv1 = tf.contrib.layers.convolution2d(
                self.embedded_chars_expanded, 
                num_filters,
                [filter_size,embedding_size], 
                padding='VALID', 
                normalizer_fn=tf.contrib.layers.batch_norm, 
                normalizer_params={'decay': 0.9}, 
                weights_regularizer=weight_decay)
# Apply nonlinearity
h=tf.nn.relu(conv1)
```

**Comment by Nitinsiwach** [comment]
one has to add update ops to control dependencies as is the case when applying batch norm as a separate layer via tf.contrib.layers.batch_norm or not?
